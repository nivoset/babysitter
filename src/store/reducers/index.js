import { set, addHours } from "date-fns";
import { calculateWageEarned } from "./calculateWage"

// Actions
export const UPDATE_START_DATE = "update_start_date";
export const UPDATE_BEDTIME_DATE = "update_bedtime_date";
export const UPDATE_END_DATE = "update_end_date";

const startDateTime = set(new Date(), { hours: 5, minutes: 0});
const bedtimeDateTime = set(new Date(), { hours: 9, minutes: 0});
const endDateTime = set(new Date(), { hours: 10, minutes: 0});

const defaultState = {
  startDateTime,
  bedtimeDateTime,
  endDateTime,
  wageEarned: calculateWageEarned(startDateTime, bedtimeDateTime, endDateTime),
}

export const rootReducer = (store = defaultState, action) => {
  switch(action.type) {
    case UPDATE_BEDTIME_DATE: 
      return { 
        ...store,
        bedtimeDateTime: addHours(action.payload, -12),
      };
    case UPDATE_START_DATE:
      return { 
        ...store,
        startDateTime: addHours(action.payload, -12)
      };
    case UPDATE_END_DATE:
      return { 
        ...store,
        endDateTime: addHours(action.payload, -12)
      };
    default: return store;
  }
}

//Action generators
export const updateStartDate = (dispatch) => (startDate) => {
  dispatch({ type: UPDATE_START_DATE, payload: startDate})
}

export const updateBedtimeDate = (dispatch) => (endDate) => {
  dispatch({ type: UPDATE_BEDTIME_DATE, payload: endDate})
}

export const updateEndDate = (dispatch) => (endDate) => {
  dispatch({ type: UPDATE_END_DATE, payload: endDate})
}

//Selectors
export const selectStartDate = (state) => addHours(state.startDateTime, 12);
export const selectBedtime = (state) => addHours(state.bedtimeDateTime, 12);
export const selectEndDate = (state) => addHours(state.endDateTime, 12);
export const selectWageEarned = (store) => calculateWageEarned(store);
