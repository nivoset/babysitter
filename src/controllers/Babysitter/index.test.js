import React from "react";
import { BabysitterController } from "./index"
import { render, screen, fireEvent } from "@testing-library/react"
import { Provider } from "react-redux";
import { store } from "../../store/index"

it("Renders without crashing", async () => {
  const {asFragment } = render(<Provider store = {store}><BabysitterController /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});