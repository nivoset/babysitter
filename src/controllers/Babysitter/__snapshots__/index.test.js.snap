// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="display-items"
  >
    <div
      id="start-date"
      title="Start Time"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        format="hh:mm a"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            readonly=""
            type="text"
            value="05:00 PM"
          />
        </div>
      </div>
    </div>
    <div
      id="bedtime"
      title="Bed Time"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        format="hh:mm a"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            readonly=""
            type="text"
            value="09:00 PM"
          />
        </div>
      </div>
    </div>
    <div
      id="end-date"
      title="End Time"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        format="hh:mm a"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            readonly=""
            type="text"
            value="10:00 PM"
          />
        </div>
      </div>
    </div>
    <div
      id="display"
      title="Display wage earned"
    >
      wages earned= $56
    </div>
  </div>
</DocumentFragment>
`;
